### PR TITLE
Fix repos downgrade in transactional_server test

### PR DESCRIPTION
**Pull request by OleksandrOrlov**
There was a revert to the previous snapshot between installation and
shutdown.

That happened in 'force_scheduled_tasks', that triggers all system
timers - in case of MicroOS that implies the transactional-update.timer,
which in turn runs `zypper dup`. At this time though, the repositories
have NOT been switched to the snapshot to be tested, but still point to
download.o.o (aka the last published snapshot) - we thus DOWNGRADE the
system to a previous snapshot and then pass THIS on to the child test.

The commit puts correct repos for the create hdd job.

**The fix suggested in the poo#86078 has fixed the issue: Proper snapshot version is used now: https://openqa.opensuse.org/tests/1607576#step/boot_to_desktop/2**

- Related ticket: https://progress.opensuse.org/issues/86078
- Verification runs: 
   - create_hdd_transactional_server: https://openqa.opensuse.org/tests/1607471
   - extra_tests_transactional_server: https://openqa.opensuse.org/tests/1607576

### CODE REVIEW

File: schedule/yast/transactional_server/create_hdd_transactional_server.yaml [hunk 1]
```diff
@@ -67,8 +67,11 @@ schedule:
     - "{{reconnect_mgmt_console}}"
     - "{{grub_test}}"
     - installation/first_boot
-    - console/hostname
```
Comment (ggardet) [inline schedule/yast/transactional_server/create_hdd_transactional_server.yaml:70]: Any reason to move `console/hostname`?

Comment (OleksandrOrlov) [reply schedule/yast/transactional_server/create_hdd_transactional_server.yaml:70]: I moved it [after the modules with zypper](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/e390cd321477d419eeb1fdcfccb3a33dc8bc9d9b/schedule/yast/transactional_server/create_hdd_transactional_server.yaml#L74) just to be consistent with the similar test suite [create_hdd_gnome](https://openqa.opensuse.org/tests/1602616). I guess it will not harm.

**Comment by jknphy** [comment]
LGTM

**Review by rwx788 — CHANGES_REQUESTED** [review]
`schedule/yast/transactional_server/create_hdd_transactional_server.yaml` is used in SLES test where this change will break tests. We need to create separate schedule for openSUSE.

**Comment by OleksandrOrlov** [comment]
> `schedule/yast/transactional_server/create_hdd_transactional_server.yaml` is used in SLES test where this change will break tests. We need to create separate schedule for openSUSE.

Added.

**Comment by ggardet** [comment]
> > `schedule/yast/transactional_server/create_hdd_transactional_server.yaml` is used in SLES test where this change will break tests. We need to create separate schedule for openSUSE.
> 
> Added.

Why should create another file? Cannot it be handle with conditionals if something is different? We should avoid to create separate files.

File: schedule/yast/transactional_server/create_hdd_transactional_server_opensuse.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+name: create_hdd_transactional_server_opensuse
+description: >
+  Installation of a Transactional Server which uses a read-only
+  root filesystem to provide atomic, automatic updates of a
+  system without interfering with the running system.
+vars:
+  SCC_ADDONS: tsm
```
Comment (rwx788) [inline schedule/yast/transactional_server/create_hdd_transactional_server_opensuse.yaml:8]: I guess we can get rid of this variable?

File: schedule/yast/transactional_server/create_hdd_transactional_server.yaml [hunk 3]
```diff
@@ -29,33 +29,18 @@ conditional_schedule:
...
-    user_settings_root:
-        DISTRI:
-            sle:
-            - installation/user_settings_root
 schedule:
     - installation/bootloader_start
     - installation/welcome
-    - "{{license_reg_scc_addon}}"
-    - "{{online_repos_inst_mode_logpackes}}"
+    - installation/accept_license
```
Comment (rwx788) [inline schedule/yast/transactional_server/create_hdd_transactional_server.yaml:35]: What is the motivation of this change?
I also would ask to put it in a separate commit.

Comment (rwx788) [reply schedule/yast/transactional_server/create_hdd_transactional_server.yaml:35]: Explained in person ;)

**Review by rwx788 — APPROVED** [review]

**Comment by OleksandrOrlov** [comment]
> > > `schedule/yast/transactional_server/create_hdd_transactional_server.yaml` is used in SLES test where this change will break tests. We need to create separate schedule for openSUSE.
> > 
> > 
> > Added.
> 
> Why should create another file? Cannot it be handle with conditionals if something is different? We should avoid to create separate files.

@ggardet Sorry, but why should we avoid creating separate files?:) AFAIK, conditional feature was added as a compromise when just slight changes in the test suite required, e.g. related to the architecture features. In case it is opensuse and SLE differences, I strongly believe we should use separate files, as these test suites contain a lot of different steps. YAML schedule was created to avoid complex conditionals to not have the issue with readability, that we had previously with main.pm. So, the schedule there should be as simple as possible, and splitting by distributions allows to reach that easily.

The problem here is that it was not split before and I didn't notice that, but thanks to @rwx788 for pointing me out.

**Comment by rwx788** [comment]
+1 to the statement above. What we aimed is that you can clearly see which modules will be scheduled without invoking any of openQA commands. `main.pm` sin was that it provided flexibility which was not controlled and it became too much for SLES/Leap/TW. Conditional schedule is a compromise, but also might lead to unreadable yamls with tons of conditions.
Therefore, my recommendation would be, if you want to have code driven flexible scheduling, `main.pm` is a way to do that. yaml does it's only purpose and suits for dumb cases to map schedules 1 to 1. We do not fool oursleves anymore by saying "it's same test" when backend is different or it's totally different distribution.
Unfortunately, I could not find related progress ticket in 5 minutes where we had all of those discussions.

Hope this information helps a bit to understand our motivation. Cheers ;)

**Comment by DimStar77** [comment]
Third breaks transactional update:

https://openqa.opensuse.org/tests/1610975

The move off license accept is faulty

**Review by rwx788 — COMMENTED** [review]
This is indeed wrong schedule, as now we have a separate one for openSUSE.

**Comment by DimStar77** [comment]
> This is indeed wrong schedule, as now we have a separate one for openSUSE.

Will you be having the honors of adjusting it please?

**Comment by rwx788** [comment]
> > This is indeed wrong schedule, as now we have a separate one for openSUSE.
> 
> Will you be having the honors of adjusting it please?

Sure, I will do that, as seems @OleksandrOrlov is not yet available. I will also retigger failed job.

**Comment by rwx788** [comment]
@DimStar77 I've noticed that now we have  `transactional_server` test suite, and `create_hdd_transactional_server` with `extra_tests_transactional_server`. Sounds like we should get rid of `transactional_server`, shall I do it?
